packages/server-ct/src/project-ct.ts

```diff
@@ -17,12 +18,39 @@ export class ProjectCt extends ProjectBase<ServerCt> {
     return 'ct'
   }
 
+  /**
+   * override default viewport for component testing
+   * 1. if user specified viewport values in their cypress.json, use those.
+   * 2. otherwise, use 500/500 by default.
+   */
+  addComponentTestingUniqueDefaults (cfg: Record<string, unknown>) {
```

Comment:
Maybe we can override the defaults itself, instead of trying to override users config in runtime? What if user will manually set 1000x500 resolution?

Or this is doing exactly that? If yes, let's rename is cfg to defaults -- I was a little bit confused :)

Reply:
I tried to do exactly this but it's quite a lot of code to change. If we think this is worth doing now, happy to do so.

The reason this is hard is defaults are [defined here](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/config_options.ts) and accessed [here](https://github.com/cypress-io/cypress/blob/4604f24e11862a7bb083c06a812565a833c78a7f/packages/server/lib/config.js#L46), then they are used throughout that entire file. At that point, we don't know if we are in component testing mode or e2e mode - so we would need to do the following:

1. have a `baseOptions` (likely should be named `baseDefaults`, but it's called `options` here)
2. create `e2eOptions` and `ctOptions`
3. pass through the current mode *every time* we access options

This requires a ton of changes throughout the `server` package. What do you think? I would like to make this change, it's definitely more work than this ticket was initially scoped for.